src/Rules/Violations.php

```diff
@@ -106,17 +106,40 @@ public function toArray(): array
         return $this->violations;
     }
 
-    public function remove(self $violations): void
+    /**
+     * @param Violations $violations Known violations from the baseline
+     * @param bool $ignoreBaselineLinenumbers If set to true, violations from the baseline are ignored for the same file even if the line number is different
+     */
+    public function remove(self $violations, bool $ignoreBaselineLinenumbers = false): void
     {
+        if ($ignoreBaselineLinenumbers) {
+            $comparisonFunction = static function (Violation $a, Violation $b): int {
+                if (($a->getFqcn() === $b->getFqcn()) && ($a->getError() === $b->getError())) {
```

Comment:
Can we move this into a specific little function with an explicit name?
I know that it's a clear meaning but what do you think about readability?
Can we improve the readability of this code in your opinion?

Reply:
ah indeed. extracted it.